tests/components/analytics/test_analytics.py

```diff
@@ -0,0 +1,250 @@
+"""The tests for the analytics ."""
+from unittest.mock import patch
+
+import aiohttp
+
+from homeassistant.components.analytics.analytics import Analytics
+from homeassistant.components.analytics.const import (
+    ANALYTICS_ENDPOINT_URL,
+    ATTR_PREFERENCES,
+    AnalyticsPreference,
+)
+from homeassistant.components.hassio.const import DOMAIN as HASSIO_DOMAIN
+from homeassistant.const import __version__ as HA_VERSION
+
+MOCK_HUUID = "abcdefg"
+
+
+async def test_no_send(hass, caplog, aioclient_mock):
+    """Test send when no prefrences are defined."""
+    aioclient_mock.post(ANALYTICS_ENDPOINT_URL, status=200)
+    analytics = Analytics(hass)
+    await analytics.load()
+    assert analytics.preferences == []
+
+    with patch("homeassistant.helpers.instance_id.async_get", return_value=MOCK_HUUID):
+        await analytics.send_analytics()
+    assert "Nothing to submit" in caplog.text
+    assert len(aioclient_mock.mock_calls) == 0
+
+
+async def test_load_with_supervisor_diagnostics(hass, aioclient_mock, hassio_handler):
+    """Test loading with a supervisor that has diagnostics enabled."""
+    aioclient_mock.get(
+        "http://127.0.0.1/supervisor/info",
```

Comment:
We should test the supervisor api in the hassio integration. Here we should just mock out the hassio integration public functions.